keyboards/system76/system76_ec.c

```diff
@@ -0,0 +1,412 @@
+/*
+ *  Copyright (C) 2021  System76
+ *  Copyright (C) 2021  Jimmy Cassis <KernelOops@outlook.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include "dynamic_keymap.h"
+#include "raw_hid.h"
+#include "rgb_matrix.h"
+#include "version.h"
+
+enum Command {
+    CMD_PROBE         = 1,   // Probe for System76 EC protocol
+    CMD_BOARD         = 2,   // Read board string
+    CMD_VERSION       = 3,   // Read version string
+    CMD_RESET         = 6,   // Reset to bootloader
```

Comment:
This command should be removed.

Reply:
Can you elaborate? These commands are equivalently mapped to the Keyboard Configurator application and they are sent through HID.

Reply:
Allowing the keyboard to be put into bootloader mode by the host is a security concern. We have already removed code that does this in a couple places before, and we cannot merge this PR until this functionality is removed.

Reply:
Actually there seems to be a security check there — `CMD_RESET` does nothing if the `bootloader_unlocked` variable is not set, and that variable gets set in `system76_ec_unlock()`, which is called from `process_record_kb()` when handling the `RESET` keycode. (The same `bootloader_unlocked` variable also *disables* RGB control commands, presumably to avoid interfering with the special RGB mode that indicates the unlocked bootloader state.)

But this behavior is completely different from any other keyboard in QMK. I wonder whether we should just treat all this “System76 EC” stuff as vendor-specific and just drop it in upstream QMK (eventually all that keymap and RGB control should be replaced by XAP, and for now we could have VIA for the keymap part).